Tests/SwiftParserTest/TypeTests.swift

```diff
@@ -192,4 +192,42 @@ final class TypeTests: ParserTestCase {
       """
     )
   }
+
+  func testLowercaseSelf() {
+    assertParse(
+      "1️⃣self2️⃣",
+      { TypeSyntax.parse(from: &$0) },
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "unexpected 'self' keyword before type"),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "expected type", fixIts: ["insert type"]),
+      ],
+      fixedSource: "<#type#>self"
```

Comment:
`<#type#>.self`?

Reply:
We consume `self` as unexpected here. I’m not sure if `<#type#>.self` is really any more useful than what we currently produce, so not sure if it’s worth adding a special case for this recovery. 

To put it into context, if you write `let a: self`, I don’t think that `let a: <#type#>.self` is a meaningful recovery strategy. We could suggest wrapping `self` in backticks here, but I’m 100% sure that’s not what the developer intended either.